### PR TITLE
[Flang] Switch to common::visit more call sites

### PR DESCRIPTION
**Pull request by alexander-shaposhnikov**
Switch to common::visit more call sites + git clang-format --style=llvm.

Test plan: ninja check-all

### CODE REVIEW

File: flang/include/flang/Evaluate/tools.h [hunk 1]
```diff
@@ -448,12 +448,12 @@ struct ExtractSubstringHelper {
 
   template <typename T>
   static std::optional<Substring> visit(const Designator<T> &e) {
-    return std::visit([](auto &&s) { return visit(s); }, e.u);
+    return Fortran::common::visit([](auto &&s) { return visit(s); }, e.u);
```
Comment (klausler) [inline flang/include/flang/Evaluate/tools.h:451]: The `Fortran::` namespace is redundant here, so in Evaluate and Semantics I don't use it.

Comment (alexander-shaposhnikov) [reply flang/include/flang/Evaluate/tools.h]: Done

**Comment by alexander-shaposhnikov** [comment]
@klausler gentle ping

**Review by klausler — APPROVED** [review]

**Review by jeanPerier — APPROVED** [review]
LGTM, thanks

**Pull request merged**